compal/__init__.py

```diff
@@ -926,6 +927,204 @@ def turn_off(self, debug=False):
         self.update_wifi_settings(settings, debug)
 
 
+class WifiGuestNetworkSettings(object):
+    """
+    Configures the WiFi guest network settings
+    """
+
+    def __init__(self, modem):
+        # The modem sometimes returns invalid XML when 'strange' values are
+        # present in the settings. The recovering parser from lxml is used to
+        # handle this.
+        self.parser = etree.XMLParser(recover=True)
+
+        self.modem = modem
+
+    @property
+    def wifi_guest_network_settings_xml(self):
+        """
+        Get the current wifi guest network settings as XML
+        """
+        xml_content = self.modem.xml_getter(GetFunction.WIRELESSGUESTNETWORK, {}).content
+        return etree.fromstring(xml_content, parser=self.parser)
+
+    @staticmethod
+    def band_guest_networks(xml, band):
+        """
+        Get the wifi guest network settings for the given band (2g, 5g)
+        """
+        assert band in ("2g", "5g",)
+
+        all_interfaces = list()
+        interfaces = xml.iter("Interface" + ("" if band == "2g" else band.upper()))
+        for interface in interfaces:
+            def xml_value(attr, coherce=True):
```

Comment:
If I read this correctly, only `interface` is used from the scope the function is defined in. This means that you could define this function on the class (for example `_xml_value(self, interface, attr, coherce=True)`) and pass in the interface instead of in the loop.

Reply:
Your absolut right, I moved the function to class level.